irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -119,6 +120,20 @@ namespace iroha {
               std::make_unique<soci::session>(*connection_)));
     }
 
+    boost::optional<std::shared_ptr<QueryExecutor>>
+    StorageImpl::createQueryExecutor() const {
+      std::shared_lock<std::shared_timed_mutex> lock(drop_mutex);
+      if (not connection_) {
+        log_->info("connection to database is not initialised");
+        return boost::none;
+      }
+      return boost::make_optional<std::shared_ptr<QueryExecutor>>(
+          std::make_shared<PostgresQueryExecutor>(
+              std::make_unique<soci::session>(*connection_),
```

Comment:
Is it possible to use some sort of lazy connection here, so that it is not required to check `connection_`?